Fix TS critters

reaperrr opened the pull request:
Fixes bit-rotten wander delays (oopsie...), Fiend selection box, and adult visc weapon and attack anim.

reaperrr committed:
Fix Tiberian Fiend selection box

reaperrr committed:
Slightly tweak default TS infantry selection box

reaperrr committed:
Fix and tweak TS critter wander delays

The property name has bit-rotted, and the values of the Tiberian Fiend were too low/short, resulting in too much hectic activity.

reaperrr committed:
Fix large TS visceroid attack and enable attack animation

Unfortunately the animation sprite starts at an unusual facing, so we have to work around that by manually fixing the order. Using Combine is easier than listing a whopping 40 frames in the correct order.

GraionDilach reviewed (approved):
Looks alright to me. :+1:

pchote commented on mods/ts/sequences/infantry.yaml:
```diff
@@ -526,12 +526,37 @@ vislrg:
-	muzzle: vislgatk # TODO: unused
+	attack:
+		Combine:
+			vislgatk:
```
This looks mostly redundant... why can't you merge the first 7 into a single `Start: 5` `Length: 35`?

pchote replied:
Also yay for hardcoded (or unused?) inconsistent frame mappings.

reaperrr replied:
> why can't you merge the first 7 into a single Start: 5 Length: 35?

I just didn't think of that.

pchote commented on mods/ts/rules/civilian-infantry.yaml:
```diff
@@ -270,13 +272,16 @@ VISC_LRG:
 		HP: 500
 	Armament:
 		Weapon: SlimeAttack
+		FireDelay: 10
```
Neither these nor `doggie`s will attack me when placed as creeps in the map editor.  What's going on?

reaperrr replied:
Are you sure you selected Creeps as owner and the Multi players as enemies?

They don't attack on Tiberium Garden Redux for example because the Owner is set to Neutral (and yes, changing that to Creeps worked for me).

Either way, I guess I should fix the official maps with this PR.

pchote replied:
100% sure:

![screen shot 2017-01-15 at 15 39 10](https://cloud.githubusercontent.com/assets/167819/21963844/135cb204-db39-11e6-9167-f753e3a53365.png)

```
	Actor235: doggie
		Owner: Creeps
		Location: 46,22
		SubCell: 2
		Facing: 92
	Actor236: visc_lrg
		Owner: Creeps
		Location: 48,25
		Facing: 92
	Actor237: visc_sml
		Owner: Creeps
		Location: 43,23
		Facing: 92
```

pchote replied:
Changing the owners in t_garden also doesn't work for me.

reaperrr replied:
doggie should really work, it does on TGR with Owner fixed.

visc_lrg should at least react to attacks, it might not attack on its own due to low scan radius.

I can't reproduce that issue with a new map either, sorry.

reaperrr replied:
> Changing the owners in t_garden also doesn't work for me.

o.O

Are we maybe running into a Mono bug here?

pchote replied:
Can you link to a replay that shows off these critters attacking on the latest version of this branch?  If it desyncs then we know its a bug.

reaperrr replied:
Uploaded replay to http://tdx.ppmsite.com/OpenRA-2017-01-15T184844Z.zip

I first sent a bike to a visceroid, next a buggy to a tib fiend, in both cases they actively attacked my vehicles.

pchote replied:
They do indeed attack on your replay, but still not when I play.  I must be doing something wrong...?  Here's a replay where I chase around and then very slowly kill an adult visceroid with a buggy without retaliation: https://www.dropbox.com/s/my2kaw29op4hthu/OpenRA-2017-01-15T202124Z.orarep?dl=0

reaperrr replied:
Sorry for asking the obvious, but I guess you did make sure you updated to the latest version (with the map fixes) of this branch before creating that replay?

pchote replied:
Quite sure, yeah.  It is on the same version that i played back your replay with, and that would have desynced if the map owners disagreed

pchote replied:
I was able to track down the problem: the critters don't reveal shroud, so they are not able to see anything around them.  This only worked for you because you disabled fog and shroud in the lobby options!

reaperrr replied:
Oh wow... well, I'm glad it's easy to fix.

reaperrr committed:
Fix critter owner on shipping maps

reaperrr committed:
Give adult visceroid a larger target scan radius

reaperrr commented:
Updated.

Fixed Owner on shipping maps and increased visceroid AutoTarget scan radius.

reaperrr committed:
Fix TS Tib Fiend and adult visceroid sight range

Otherwise they won't attack with shroud enabled!

reaperrr committed:
Fix adult visceroid armor type

As per original rules.ini.

reaperrr commented:
Updated.

- Simplified visc attack sequence as suggested
- Gave armed critters a reveal range of 4 cells (original game had 4 for doggie and 0 for visc, but TS AI ignored shroud anyway)
- Set visc AutoTarget to 5 cells (as per original GuardRange=5)
- Fixed adult visc amor type as per original rules.ini

pchote reviewed (approved):
Looks good, but one final nit: we probably don't want these critters gaining veterancy?

reaperrr committed:
Remove veterancy from Tiberian Fiend

Matches original game as well.

reaperrr commented:
> Looks good, but one final nit: we probably don't want these critters gaining veterancy?

You're right, fixed.

abcdefg30 commented:
Needs a rebase, though.

reaperrr commented:
Rebased.

pchote commented on mods/ts/sequences/infantry.yaml:
```diff
@@ -526,12 +526,19 @@ vislrg:
-		ShadowStart: 40
 		Tick: 80
 		Offset: 0, 0, 16
+#		ShadowStart: 40 # TODO: enable shadow frames
```
Do we have a ticket for the shadow frame limitations?

reaperrr replied:
Sort of: #11803
